static/app/utils/replays/hooks/useExtractedPageHtml.tsx

```diff
@@ -0,0 +1,21 @@
+import {useQuery} from 'sentry/utils/queryClient';
+import extractPageHtml from 'sentry/utils/replays/extractPageHtml';
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+
+interface Props {
+  offsetMsToStopAt: number[];
+  replay: ReplayReader | null;
+}
+
+export default function useExtractedPageHtml({replay, offsetMsToStopAt}: Props) {
+  return useQuery(
+    ['extactPageHtml', replay],
```

Comment:
not sure what is hashing the replay to compare it later but it does look like it might access every property on the object? https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#query-keys-are-hashed-deterministically

Reply:
We've implemented `toJSON` to avoid that specific problem! It was indeed crazy without that

https://github.com/getsentry/sentry/blob/d72428194f97d82e2f6d668673175213b0e6e5f2/static/app/utils/replays/replayReader.tsx#L383